src/NServiceBus.Core/Recoverability/RecoverabilitySettings.cs

```diff
@@ -0,0 +1,230 @@
+        /// </summary>
+        public DelayedConfig Delayed { get; }
+    }
+
+    /// <summary>
+    /// Provides information about the immediate retries configuration.
+    /// </summary>
+    public struct ImmediateConfig
+    {
+        internal ImmediateConfig(int maxNumberOfRetries)
+        {
+            MaxNumberOfRetries = maxNumberOfRetries;
+        }
+
+        /// <summary>
+        /// Gets the configured maximum number of immediate retries.
+        /// </summary>
+        /// <remarks>Zero means no retries possible.</remarks>
+        public int MaxNumberOfRetries { get; }
+    }
+
+    /// <summary>
+    /// Provides information about the delayed retries configuration.
+    /// </summary>
+    public struct DelayedConfig
+    {
+        internal DelayedConfig(int maxNumberOfRetries, TimeSpan timeIncrease)
+        {
+            MaxNumberOfRetries = maxNumberOfRetries;
+            TimeIncrease = timeIncrease;
+        }
+
+        /// <summary>
+        /// Gets the configured maximum number of immediate retries.
+        /// </summary>
+        /// <remarks>Zero means no retries possible.</remarks>
+        public int MaxNumberOfRetries { get; }
+
+        /// <summary>
+        /// Gets the configured time of increase for individual delayed retries.
+        /// </summary>
+        public TimeSpan TimeIncrease { get; }
+    }
+
+    /// <summary>
+    /// Extension methods for recoverability which extend <see cref="EndpointConfiguration" />.
+    /// </summary>
+    public static class RecoverabilityEndpointConfigurationExtensions
+    {
+        /// <summary>
+        /// Configuration settings for recoverability.
+        /// </summary>
+        /// <param name="configuration">The endpoint configuration.</param>
+        public static RecoverabilitySettings Recoverability(this EndpointConfiguration configuration)
+        {
+            return new RecoverabilitySettings(configuration.GetSettings());
+        }
+    }
+
+    /// <summary>
+    /// Configuration settings for immediate retries.
+    /// </summary>
+    public class ImmediateRetriesSettings : ExposeSettings
+    {
+        internal ImmediateRetriesSettings(SettingsHolder settings) : base(settings)
+        {
+        }
+
+        /// <summary>
+        /// Configures the amount of times a message should be immediately retried after failing before escalating to second level
+        /// retries.
+        /// </summary>
+        /// <param name="numberOfRetries">The number of times to immediately retry a failed message.</param>
+        public void NumberOfRetries(int numberOfRetries)
+        {
+            Guard.AgainstNegative(nameof(numberOfRetries), numberOfRetries);
+
+            Settings.Set(Recoverability.FlrNumberOfRetries, numberOfRetries);
+        }
+
+        /// <summary>
+        /// Configures NServiceBus to not retry failed messages using the first level retry mechanism.
+        /// </summary>
+        public void Disable()
```

Comment:
Do we still need this method? I would prefer the approach: "All retries are now part of Recoverability. Recoverability cannot be disabled - failing message always ends in error queue. You can however control number of retries"